hk2-locator/src/test/java/org/glassfish/hk2/tests/locator/optional/InjectedManyTimes.java

```diff
@@ -45,6 +45,24 @@
     @Inject @Optional
     private IterableProvider<OptionalService> optionalByIterableProvider;
     
+    @Inject
+    private java.util.Optional<OptionalService> optionalByContained;
+    
+    @Inject
+    private java.util.Optional<SimpleService> simpleByContained;
+    
+    @Inject
+    private java.util.Optional<Provider<OptionalService>> optionalByProviderContained;
```

Comment:
Just out of curiosity: Would `Optional<Optional<X>>` work?

Reply:
yes it would. Added test to prove it.

Reply:
Nice :+1: